lib/build_data.sh

```diff
@@ -43,20 +43,20 @@ function build_data::set_string() {
 }
 
 # Sets a build data value for the elapsed time in seconds between the provided start time and the
-# current time, represented as a float with milliseconds precision.
+# current time, represented as a float with microseconds precision.
 #
 # Usage:
 # ```
-# local dependencies_install_start_time=$(build_data::current_unix_time_ms)
+# local dependencies_install_start_time=$(build_data::current_unix_realtime)
 # # ... some operation ...
 # build_data::set_duration "dependencies_install_duration" "${dependencies_install_start_time}"
 # ```
 function build_data::set_duration() {
 	local key="${1}"
 	local start_time="${2}"
 	local end_time duration
-	end_time="$(build_data::current_unix_time_ms)"
-	duration="$(awk -v start="${start_time}" -v end="${end_time}" 'BEGIN { printf "%.3f", (end - start)/1000 }')"
+	end_time="$(build_data::current_unix_realtime)"
+	duration="$(awk -v start="${start_time}" -v end="${end_time}" 'BEGIN { printf "%f", (end - start) }')"
```

Comment:
```suggestion
	duration="$(awk -v start="${start_time}" -v end="${end_time}" 'BEGIN { print (end - start) }')"
```

Reply:
The `print` causes less precision, eg:

```
root@d4d36b9d6a56:/# awk -v start="${start_time}" -v end="${end_time}" 'BEGIN { print (end - start) }'
13.2063
```

So it would be better to either switch back to `printf`:

```
root@d4d36b9d6a56:/# awk -v start="${start_time}" -v end="${end_time}" 'BEGIN { printf "%f", (end - start) }'
13.206290
```

Or use `OFMT` with `print`:
```
root@d4d36b9d6a56:/# awk -v start="${start_time}" -v end="${end_time}" -v OFMT="%f" 'BEGIN { print (end - start) }'
13.206290
```

Given that we don't want the trailing newline (albeit the command substitution removes it for us), lets go with `printf` again (plus it saves people having to understand `OFMT`).